tests/test_plane.h

```diff
@@ -0,0 +1,183 @@
+/*************************************************************************/
+/*                       This file is part of:                           */
+/*                           GODOT ENGINE                                */
+/*                      https://godotengine.org                          */
+/*************************************************************************/
+/* Copyright (c) 2007-2020 Juan Linietsky, Ariel Manzur.                 */
+/* Copyright (c) 2014-2020 Godot Engine contributors (cf. AUTHORS.md).   */
+/*                                                                       */
+/* Permission is hereby granted, free of charge, to any person obtaining */
+/* a copy of this software and associated documentation files (the       */
+/* "Software"), to deal in the Software without restriction, including   */
+/* without limitation the rights to use, copy, modify, merge, publish,   */
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#ifndef TEST_PLANE_H
+#define TEST_PLANE_H
+
+#include "core/math/plane.h"
+
+#include "thirdparty/doctest/doctest.h"
+
+namespace TestPlane {
+
+// Plane
+
+TEST_CASE("[Plane] Constructor methods") {
+	const Plane plane = Plane(32, 22, 16, 3);
+	const Plane plane_vector = Plane(Vector3(32, 22, 16), 3);
+	const Plane plane_copy_plane = Plane(plane);
+
+	CHECK_MESSAGE(
+			plane == plane_vector,
+			"Planes created with same values but different methods should be equal.");
+
+	CHECK_MESSAGE(
+			plane == plane_copy_plane,
+			"Planes created with same values but different methods should be equal.");
+}
+
+TEST_CASE("[Plane] Basic getters") {
+	const Plane plane = Plane(32, 22, 16, 3);
+	const Plane plane_normalized = Plane(32.0 / 42, 22.0 / 42, 16.0 / 42, 3.0 / 42);
+
+	CHECK_MESSAGE(
+			plane.get_normal().is_equal_approx(Vector3(32, 22, 16)),
+			"get_normal() should return the expected value.");
```

Comment:
Unless the assertion is not obvious or requires some preparation steps (complex), I don't find it necessary to repeat what it does in words, so I'd just use plain `CHECK` macro in those cases, the testing framework is smart enough to showcase what are the actual and expected values.

Otherwise this adds too much verbosity, in my opinion.

But I realize this is already the case as seen in some other test suites. It's fine to have those messages. Perhaps documentation should also provide an objective criteria to elaborate what kind of assertions require to use either `CHECK` or `CHECK_MESSAGE`.